src/control/server/instance_storage_rpc.go

```diff
@@ -198,6 +198,9 @@ func scanEngineBdevsOverDrpc(ctx context.Context, engine Engine, pbReq *ctlpb.Sc
 			return nil, errors.Errorf("smd %q has no ctrlr ref", sd.Uuid)
 		}
 
+		if sd.Ctrlr.DevState == ctlpb.NvmeDevState_UNPLUGGED {
```

Comment:
I wonder if would be safer to skip the controller unless its state is in a set of known-good/expected states.

Reply:
I have updated the PR to address this.